pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -478,11 +478,19 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(pt.validateRefOrSpec())
 
 	errs = errs.Also(pt.validateEmbeddedOrType())
-
+	notCustomTaskKinds := map[TaskKind]bool{
+		"":                 true,
+		NamespacedTaskKind: true,
```

Comment:
@jerop can you take a look here? I'm not sure if validation runs before or after defaulting. If the user sets "kind" = Task, didn't we decide that should be a custom task?

Reply:
if user sets kind to task:
- if apiversion is non-empty --> custom task
- if apiversion is empty --> task (because this is what we default to, we add kind but leave apiversion as empty)